app/views/session_timeout/_ping.js.erb

```diff
@@ -51,4 +66,20 @@ function success(data) {
   }
 }
 
+function keepalive() {
+  var request = new XMLHttpRequest();
+  request.open('POST', '/sessions/keepalive', true);
+  request.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
```

Comment:
Curious: Is this necessary?

Reply:
It doesn't seem like it, I tested locally and nothing broke. The code is quite old, and it seems to be related to CORS, so I've removed and will confirm if anything changes in a deployed environment